calico_node/startup/startup.go

```diff
@@ -102,6 +102,13 @@ func main() {
 		configureASNumber(node)
 	}
 
+	if nodeRef := os.Getenv("CALICO_K8S_NODE_REF"); nodeRef != "" {
+		node.Spec.OrchRefs = append(node.Spec.OrchRefs, api.OrchRef{
```

Comment:
Doesn't this mean that each time the node is rebooted, it will add an additional orchRef? 

I think it needs to check if a "k8s" orchRef has been set yet, and if not set it.

Additionally, I think we need to detect which orchestrator we're running in here. Looks like this code path runs for all orchestrators. But we only want to set it if we're running in k8s.

Reply:
Yea I agree on both points, do we have a good way to detect the orchestrator?